hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -1065,8 +1071,8 @@ def generate_new_examples(self) -> None:
                 # ConjectureResult object.
                 assert isinstance(minimal_example, ConjectureResult)
                 consecutive_zero_extend_is_invalid = 0
-                minimal_extension = ir_size(minimal_example.ir_nodes) - prefix_size
-                max_length = min(prefix_size + minimal_extension * 10, BUFFER_SIZE_IR)
+                minimal_extension = len(minimal_example.choices) - len(prefix)
+                max_length = len(prefix) + minimal_extension * 5
```

Comment:
behavior change: using `* 5` instead of `* 10`, since we're now basing the size off number of choices, not the buffer length. The argument is that a choice takes more than one byte on average. 5 is mostly arbitrary.